File: CS/GrowthWare.Framework/Model/Profiles/Interfaces/IMGroupRoleSecurity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections.ObjectModel;

namespace GrowthWare.Framework.Model.Profiles.Interfaces
{
    /// <summary>
    /// Interface for anything needed group and role security
    /// </summary>
    [CLSCompliant(true)]
    public interface IMGroupRoleSecurity
    {
        /// <summary>
        /// Roles that are directly assigned
        /// </summary>
        Collection<String> AssignedRoles
        {
            get;
        }

        /// <summary>
        /// Roles that are derived from groups
        /// </summary>
        Collection<String> DerivedRoles
        {
            get;
        }

        /// <summary>
        /// Groups that are directly assigned
        /// </summary>
        Collection<String> Groups
        {
            get;
        }
    }
}

File: VB/GrowthWare.Framework/Model/Profiles/Interfaces/IMGroupRoleSecurity.vb
Imports System.Collections.ObjectModel

Namespace Model.Profiles.Interfaces
    ''' <summary>
    ''' Interface for anything needed group and role security
    ''' </summary>
    <CLSCompliant(True)> _
    Public Interface IMGroupRoleSecurity
        ''' <summary>
        ''' Roles that are directly assigned
        ''' </summary>
        ReadOnly Property AssignedRoles() As Collection(Of String)

        ''' <summary>
        ''' Roles that are derived from groups
        ''' </summary>
        ReadOnly Property DerivedRoles() As Collection(Of String)

        ''' <summary>
        ''' Groups that are directly assigned
        ''' </summary>
        ReadOnly Property Groups As Collection(Of String)

    End Interface
End Namespace
